### ISSUE TITLE
why combined loss is better？

### ISSUE THREAD

**Issue by xubaoquan33** [natanielruiz/deep-head-pose#37]
I want to get an answer about my question. But i can't find answer from the artical.
In my opinion, the answer may be that combined loss is suitable for training models. Many comment point out that regression loss is hard for training,cls loss is easier.
So,how do you think about it?

**Comment by alvenchen** [comment]
I'm get confused too. And I try to train on AFW and HELEN dataset using pretrained model resnet34,but it seems can't get convergence, especially MSELoss.

![image](https://user-images.githubusercontent.com/828895/45083510-96b73080-b12e-11e8-9bd6-bc823050ad9a.png)

**Comment by alvenchen** [comment]
@xubaoquan33   Thanks for your explanation. I used HELEN datasets,  perhaps it's too small, I'll try AFLW later.
    And I think parameter α in combined loss maybe differ in different dataset, even could be dynamic in different epoch just like learning rate. 
    By the way, what dataset and optimizer are you using?  thanks..

**Comment by xubaoquan33** [comment]
@alvenchen 300W dataset used for training,ALWF2000 used for test. the same with the paper.

**Comment by haofanwang** [comment]
In fact, the purpose to introduce the cls loss is to relax this problem, from a regression problem to a classification problem (the regression part is more like a post processing), you know, a coarse label is usually easier to learn.

**Comment by xubaoquan33** [comment]
@haofanwang I know that.But it is not a rigorous explanation.Changing regression to bin classification  means that you think the angle distribution of dataset is MultiNoulli. If the angles in dataset has class-imbalance, the bin classification model will be bad. And in fact ,the dataset of 300W and AFLW2000,their class-imbalance is terrible .If we used these dataset to train and test our model ,we should think a question :will the model play well in the real scene?

**Comment by haofanwang** [comment]
Yes, I think about it for a while, and indeed, due to the angle usually is Normal Distribution, softmax may not be the perfect choice here. This problem need to be explored further more. Nice point!

**Comment by natanielruiz** [comment]
Good discussion and points. The paper doesn't try to explain the reason as to why the combined loss is better but hypothesizes that using the combined loss the training is more stable and observes that test error is reduced maybe due to some sort of regularization since these datasets are easy to overfit to. Just some ideas.
